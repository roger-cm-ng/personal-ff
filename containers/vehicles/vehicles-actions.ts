import { Dispatch } from 'redux';
import { getVehicles } from '../../helpers/resources';
import { inject } from './vehicles-utils';
import { ERROR_RAISED } from '../error-handler/error-handler-actions';
import ErrorRestart from '../../components/error-content/error-restart/error-restart';

export const VEHICLES_FETCHED = 'VEHICLES_FETCHED';

export const fetchVehicles = () => async (dispatch: Dispatch) => {
  try {
    const payload = await getVehicles();

    const { data, status } = payload;

    if (status >= 200 && status < 300) {
      dispatch({
        type: VEHICLES_FETCHED,
        payload: inject(data)
      });
    } else {
      dispatch({
        type: ERROR_RAISED,
        component: ErrorRestart
      });
    }
  } catch (err) {
    dispatch({
      type: ERROR_RAISED,
      component: ErrorRestart
    });
  }
};
