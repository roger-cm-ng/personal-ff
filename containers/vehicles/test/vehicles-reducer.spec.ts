import vehiclesReducer from '../vehicles-reducer';
import { VEHICLES_FETCHED } from './../vehicles-actions';
import {
  VEHICLES_MOCK
} from './vehicles-mock-data';

describe('vehicles reducer', () => {
  const initialState = () => vehiclesReducer([], {
    type: VEHICLES_FETCHED,
    payload: []
  });

  const acquiredState = () => vehiclesReducer(initialState(), {
    type: VEHICLES_FETCHED,
    payload: VEHICLES_MOCK
  });

  it('should return initial state', () => {
    expect(initialState()).toHaveLength(0);
  });

  it('should update the state with acquired data when type is VEHICLES_FETCHED', () => {
    expect(acquiredState()).toMatchSnapshot();
    expect(acquiredState()).toEqual(VEHICLES_MOCK);
  });
});
