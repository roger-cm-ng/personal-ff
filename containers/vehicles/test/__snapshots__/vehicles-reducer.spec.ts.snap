// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vehicles reducer should update the state with acquired data when type is VEHICLES_FETCHED 1`] = `
Array [
  Object {
    "capacity": true,
    "max_distance": 200,
    "name": "Space pod",
    "speed": 2,
    "total_no": 2,
    "usage": 0,
  },
  Object {
    "capacity": true,
    "max_distance": 300,
    "name": "Space rocket",
    "speed": 4,
    "total_no": 1,
    "usage": 0,
  },
  Object {
    "capacity": true,
    "max_distance": 400,
    "name": "Space shuttle",
    "speed": 5,
    "total_no": 1,
    "usage": 0,
  },
  Object {
    "capacity": true,
    "max_distance": 600,
    "name": "Space ship",
    "speed": 10,
    "total_no": 2,
    "usage": 0,
  },
]
`;
